File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height):
        l=0
        b=len(height)-1
        ar=0
        while l<b:
            w=b-l
            m=min(height[l], height[b])
            cur= w*m
            ar= max(ar,cur)
            if height[l]<height[b]:
                l+=1
            else:
                b-=1
        return ar
if __name__ == "__main__":
    sol = Solution()

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0;
        int b=height.size()-1;
        int res=0;
        while(l<b){
            int w=b-l;
            int m=min(height[l], height[b]);
            int curr=w*m;
            res=max(res,curr);
            if (height[l]<height[b]){
                l+=1;
            }
            else{
                b-=1;
            }
        }
        return res;
    }
};
